tdp/core/deployment/deployment_iterator.py

```diff
@@ -111,20 +110,19 @@ def __init__(
 
     def __next__(
         self,
-    ) -> Optional[ProcessOperationFn]:
+    ) -> tuple[OperationModel, Optional[ProcessOperationFn]]:
         try:
             while True:
                 operation_rec = next(self._iter)
 
                 # Return early if deployment failed
                 if self.deployment.state == DeploymentStateEnum.FAILURE:
                     operation_rec.state = OperationStateEnum.HELD
-                    return
+                    return operation_rec, None
 
                 operation_rec.state = OperationStateEnum.RUNNING
 
-                return partial(self._process_operation_fn, operation_rec)
-
+                return operation_rec, self._process_operation_fn
```

Comment:
Why removing the `partial` ?

Reply:
It seems better if the caller can pass the operation as an argument directly instead of pre-passing it via partial.

Reply:
No the caller can give his own operation which he must not be able to. Only the operation provided by the iterator must be ran.

Reply:
I reverted it back to the previous version so that the operation_rec of the iterator is directly integrated.